### ISSUE TITLE
Window transparency doesn't work on Windows 10

### ISSUE THREAD

**Issue by sambercx** [tomaka/winit#260]
The ```DwmEnableBlurBehindWindow``` method for making windows transparent on Windows doesn't seem to work on Windows 10 anymore. On Windows 10 the ```DwmExtendFrameIntoClientArea``` function can be called with the margins set to -1 to make it fully transparent.

EDIT: The new method only works if the window decorations are disabled.

**Comment by LYP951018** [comment]
I'm trying to implement this by `GetProcAddress` in runtime. But should we cache `kernel32` module handle in winit? Otherwise we may need to call `LoadLibrary`every time the method is called.

**Comment by retep998** [comment]
Use `lazy_static` to lazily load `kernel32.dll` and keep the handle to it around and do the same for each of the function pointers. Never let go.

**Comment by Lisoph** [comment]
I will attempt to fix this as part of #568.

**Comment by subnomo** [comment]
I've enabled transparency using layered windows. It works with window decorations, as well. See my patch:

```diff
diff --git a/src/platform/windows/window.rs b/src/platform/windows/window.rs
index 259376f4..a227c090 100644
--- a/src/platform/windows/window.rs
+++ b/src/platform/windows/window.rs
@@ -905,6 +905,10 @@ unsafe fn init(
         ex_style |= winuser::WS_EX_NOREDIRECTIONBITMAP;
     }
 
+    if attributes.transparent {
+        ex_style |= winuser::WS_EX_LAYERED;
+    }
+
     // adjusting the window coordinates using the style
     winuser::AdjustWindowRectEx(&mut rect, style, 0, ex_style);
 
@@ -959,6 +963,11 @@ unsafe fn init(
                                               format!("{}", io::Error::last_os_error()))));
         }
 
+        if attributes.transparent {
+            let opacity = 100; // transparency value 0-255
+            winuser::SetLayeredWindowAttributes(handle, 0x0, opacity, winuser::LWA_ALPHA);
+        }
+
         WindowWrapper(handle)
     };
```
